pyccel/parser/semantic.py

```diff
@@ -829,16 +829,11 @@ def _visit_str(self, expr, **settings):
         return repr(expr)
 
     def _visit_Slice(self, expr, **settings):
-        args = list(expr.args)
-        if args[0]:
-            args[0] = self._visit(args[0], **settings)
+        start = self._visit(expr.start) if expr.start else expr.start
+        end = self._visit(expr.end) if expr.end else expr.end
+        step = self._visit(expr.step) if expr.step else expr.step
```

Comment:
Please just end these lines with `else None`, which I think is much more explicit and clearer to read.

Reply:
By the way, can we be sure that `expr.start` and `expr.end` are never equal to `0` (of type `int`)?
I'm just worried that we would get `None` in that case...